pkg/init/backend/interactive.go

```diff
@@ -51,22 +53,36 @@ func (o *InteractiveBackend) Validate(flags map[string]string, fs filesystem.Fil
 
 func (o *InteractiveBackend) SelectDevfile(ctx context.Context, flags map[string]string, _ filesystem.Filesystem, _ string) (*api.DetectionResult, error) {
 	result := &api.DetectionResult{}
-	devfileEntries, _ := o.registryClient.ListDevfileStacks(ctx, "", "", "", false, false)
-
-	langs := devfileEntries.GetLanguages()
-	state := STATE_ASK_LANG
+	var devfileEntries registry.DevfileStackList
+	state := STATE_ASK_ARCHITECTURES
 	var lang string
+	archs := []string{"amd64"}
 	var err error
 	var details api.DevfileStack
 loop:
 	for {
 		switch state {
 
+		case STATE_ASK_ARCHITECTURES:
+			archs, err = o.askerClient.AskArchitectures(knownArchitectures, archs)
+			if err != nil {
+				return nil, err
+			}
+			state = STATE_ASK_LANG
```

Comment:
```
[...]
? Select architectures to support: 
? Select language: .NET
? Select project type:  [Use arrows to move, type to filter]
> .NET 5.0 (dotnet50, registry: StagingRegistry)
  .NET 5.0 (dotnet50, registry: DefaultDevfileRegistry)
  .NET 6.0 (dotnet60, registry: StagingRegistry)
  .NET 6.0 (dotnet60, registry: DefaultDevfileRegistry)
  .NET Core 3.1 (dotnetcore31, registry: StagingRegistry)
  .NET Core 3.1 (dotnetcore31, registry: DefaultDevfileRegistry)
  ** GO BACK **
```

In the `Select architectures to support` prompt, what does it mean to select nothing? I would assume no Devfile should be proposed if the user selects nothing, because it means to me "select a Devfile not compatible with any of the supported architectures", which basically should not be allowed.
So would it be possible to disable the "none" selection, so the user selects at least one architecture? Or prompt again the user to select an architecture if the list is empty?
What do you think?

Reply:
In reality, it is more a "filter" than a "support". When no one is selected, the operation done is that no filter is applied.

Reply:
Ah, that makes sense then. In this case, maybe we can rename the prompt into something like: `Filter by supported architectures`?

Reply:
What about `Select architectures to filter on:`?

Reply:
Sounds good too, as long as it is clear that we are filtering, I think ;-)